### PR TITLE
Switch from request to request-shim

### PR DESCRIPTION
**Pull request by jason-fox**
Remove dependency on the `request` library, replace with [got](https://www.npmjs.com/package/got) - partial fix for https://github.com/telefonicaid/iotagent-node-lib/issues/858 

This is mainly a shim for the unit tests,~but since there is a new promise within the `HTTPBinding` - one of the tests `commandsPolling-test.js` is no longer 100% compliant (which is due to a difference in the way the libraries deal with `process.nextTick()` ) so you'll need to check if command statuses are  being updated properly  - it works with a manual check.~

### CODE REVIEW

**Comment by jason-fox** [comment]
From [Coveralls](https://coveralls.io/github/telefonicaid/iotagent-json)

> Coveralls is down for a planned maintenance. We will be back soon, please check back in shortly and/or check @coverallsapp for updates.

File: package.json [hunk 1]
```diff
@@ -63,10 +63,9 @@
     "body-parser": "1.18.3",
     "dateformat": "3.0.3",
     "express": "~4.16.4",
-    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
+    "iotagent-node-lib": "git://github.com/jason-fox/iotagent-node-lib.git#feature/request-shim",
```
Comment (mapedraza) [inline package.json:66]: As described in https://github.com/telefonicaid/iotagent-node-lib/pull/1119#issuecomment-1000483835, this need to be addressed (reverted) as first step of the process. 

```suggestion
    "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
```

File: package.json [hunk 2]
```diff
@@ -64,7 +64,7 @@
     "dateformat": "3.0.3",
     "express": "~4.16.4",
     "iotagent-node-lib": "git://github.com/telefonicaid/iotagent-node-lib.git#master",
-    "logops": "2.1.0",
+    "logops": "2.1.2",
```
Comment (fgalan) [inline package.json:67]: We are fixing this in master through a separate PR in each IOTA (e.g. https://github.com/telefonicaid/iotagent-json/pull/608). I'd suggest to wait for its merge, then upgrade your PRs synching from master.

Comment (jason-fox) [reply package.json:67]: Merges completed.

**Comment by fgalan** [comment]
@jason-fox I'm afraid a small merge conflict in CNR has appeared and precludes merging. Could you solve it, please?

**Comment by jason-fox** [comment]
> @jason-fox I'm afraid a small merge conflict in CNR has appeared and precludes merging. Could you solve it, please?

Done (in both cases)

**Pull request merged**